### ISSUE TITLE
One little question

### ISSUE THREAD

**Issue by JHeLiu** [aspnet/Extensions#1358]
NET Framework is used in my **`System.Configuration`** to read the Configuration file, which Microsoft official, it is very convenient, of course, the document is very clear, in
`https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/ `says is very detailed

But now that I'm moving on to.net CORE, I find that this library looks like it's changed, and I actually have a big curiosity about why.net CORE migrations aren't done in the same way as the.net Framework.I just want to hear why? I mean no harm, of course. Because I want to read it in the library, or in the console

**Comment by rynowak** [comment]
Hi @JHeLiu 

The `System.Configuration` library at the core is tied to a few old ideas:
- There is a hierarchy of configuration files that affect the app - including one as part of the OS
- Sytem.Configuration is based on XML
- System.Configuration is highly structured format that doesn't map well to other data sources

I think our main desire was to make something flexible that can work with a variety of configuration data sources (environment variables, command line args, cloud secret storage, various file formats). It would hard and confusing to try and fit these into System.Configuration's view of the world.

**Comment by JHeLiu** [comment]
@rynowak 
Thanks for your answer, I also saw.net CORE in the.net Framework
The **System.Configuration** library has been migrated.In fact, I also hope that there is a help document, a simple document description can be, such as running in the console, running in the class library,.net Framework help document is written very good, it is best to hang in Microsoft's official, this is great.I'm sure other developers would love to do that,
I found that Microsoft had done many interesting projects, but they were not well promoted. Many of them were dead. Unfortunately,
There is a small hope that the government will come up with a high concurrency big data type of enterprise solution, which can be divided into community and enterprise versions. In the past few years, the WEB has been unable to compete with JAVA and.net solutions.Of course I don't know this desire can realize ^_^, thank you for your answer.
